include/logging/log_msg2.h

```diff
@@ -34,6 +34,22 @@ typedef uint64_t log_timestamp_t;
 typedef uint32_t log_timestamp_t;
 #endif
 
+/* Determine if __auto_type is supported. If not then runtime approach must always
+ * be used.
+ *
+ * Supported by:
+ * - GCC 4.9.0 https://gcc.gnu.org/gcc-4.9/changes.html
+ * - Clang 3.8
+ */
+#ifndef CONFIG_LOG2_ALWAYS_RUNTIME
+#if !defined(__cplusplus) && \
+	!(((__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__) >= 40900) || \
+	((__clang_major__ * 10000 + __clang_minor__ * 100 + __clang_patchlevel__) >= 30800))
+#define CONFIG_LOG2_ALWAYS_RUNTIME 1
+#error
```

Comment:
? What is the point of defining something if we immediately #error anyway ?

Reply:
that was left unintentionally, fixed.